lib/pbench/server/cache_manager.py

```diff
@@ -646,6 +733,29 @@ def extract(tarball_path: Path, path: str) -> Inventory:
             tarball_path, f"Unexpected error from {tar_path}: {error_text!r}"
         )
 
+    def stream(self, path: str) -> Inventory:
+        """Return a cached inventory file as a binary stream
+
+        Args:
+            path: Relative path of a regular file within a tarball.
+
+        Returns:
+            An Inventory object encapsulating the file stream and the cache
+            lock. The caller is reponsible for closing the Inventory!
```

Comment:
It would be good if this text were explicit about the state of the lock:  a normal result returns the `Inventory` object with the cache lock locked; in the event of an error, the lock is unlocked when the exception is raised.

(Does Python guarantee that the expressions passed as arguments are evaluated from right to left?  That is, are we sure that when we call `lock.keep()` we've already returned successfully from `artifact.open()`?  It would probably be better to call `artifact.open()` before the invocation of the c'tor, just to make it explicit...and, we need to ensure that the c'tor is trivial and cannot itself raise any exception...otherwise, we would need to defer the call to `lock.keep()` until the end of the c'tor (e.g., by passing in the function and having the c'tor itself call it....)